metric/metric.go

```diff
@@ -2,26 +2,161 @@ package metric
 
 import (
 	"strconv"
+	"strings"
+	"time"
 
 	"github.com/TwiN/gatus/v3/core"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
 var (
+	namespace string
 	// This will be initialized once PublishMetricsForEndpoint.
 	// The reason why we're doing this is that if metrics are disabled, we don't want to initialize it unnecessarily.
-	resultCount *prometheus.CounterVec = nil
+	resultCount         *prometheus.CounterVec
+	resultSuccessGauge  *prometheus.GaugeVec
+	resultDurationGauge *prometheus.GaugeVec
+
+	resultDNSReturnCode *prometheus.CounterVec
+
+	resultTCPConnected *prometheus.CounterVec
+
+	resultStartTLSConnected                          *prometheus.CounterVec
+	resultStartTLSSSLLastChainExpiryTimestampSeconds *prometheus.GaugeVec
+
+	resultTLSConnected                          *prometheus.CounterVec
+	resultTLSSSLLastChainExpiryTimestampSeconds *prometheus.GaugeVec
```

Comment:
Is there a reason why two of the variables mention only TLS, while the other two mention TLS and SSL?

Reply:
I follow this naming convention ```result{Type}SSSLLastChainExpiryTimestampSeconds``` for naming.
But after add a type label, I think the naming can be changed to resultSSLLastChainExpiryTimestampSeconds.